opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java

```diff
@@ -142,6 +149,80 @@ private void updateEventAndRequiredStepsField(String baseEntityId, PartialContac
         }
     }
 
+    private void processTasks(JSONObject formObject) {
+        try {
+            String encounterType = formObject.getString(ConstantsUtils.JsonFormKeyUtils.ENCOUNTER_TYPE);
+            if (formObject.has(ConstantsUtils.JsonFormKeyUtils.ENCOUNTER_TYPE) && StringUtils.isNotBlank(encounterType) && ConstantsUtils.JsonFormUtils.ANC_TEST_ENCOUNTER_TYPE.equals(encounterType)) {
+                JSONObject dueStep = formObject.optJSONObject(JsonFormConstants.STEP1);
+                if (dueStep != null && dueStep.has(JsonFormConstants.STEP_TITLE) && ConstantsUtils.DUE.equals(dueStep.getString(JsonFormConstants.STEP_TITLE))) {
+                    JSONArray stepFields = dueStep.optJSONArray(JsonFormConstants.FIELDS);
+                    if (stepFields != null && stepFields.length() > 0) {
+                        for (int i = 0; i < stepFields.length(); i++) {
+                            JSONObject field = stepFields.getJSONObject(i);
+                            if (field != null && field.has(JsonFormConstants.IS_VISIBLE) && field.getBoolean(JsonFormConstants.IS_VISIBLE)) {
+                                JSONArray jsonArray = field.optJSONArray(JsonFormConstants.VALUE);
+                                if (jsonArray == null || (jsonArray.length() == 0) || checkTestsStatus(jsonArray)) {
+                                    saveTasks(field);
```

Comment:
Should tasks be saved if jsonArray is `null`?

Reply:
This is the creation of the tasks. A test is a valid if this 3 conditions are met 

- A test wasn't done meaning the `value` attribute is misisng from the `expansion panel` json object, hence the null check. 

- A test was done then undo thus the `value` attribute is an empty json array, hence the length check 

- A test was either marked as `not done` or `ordered`